sprintvotingapp/sprint/views.py

```diff
@@ -0,0 +1,536 @@
+import logging
+from collections import Counter
+from itertools import count
+from django.db import transaction
+from .models import Sprint, Parameter, Votes
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from .serializer import SprintSerializer, ParamSerializer, VoteSerializer
+from rest_framework.exceptions import ValidationError
+from .utils import InsertionError, AlreadyPresentException, verify_token
+from user.models import User
+
+logging.basicConfig(filename="sprint.log", filemode="w")
+
+
+class SprintC(APIView):
+    @verify_token
+    def post(self, request):
+        """
+        For Adding sprint Cycle
+        :param request:
+        :return:Response
+        """
+        try:
+            user = User.objects.get(id=request.data.get("user_id"))
+
+            if user.is_superuser:
+                serializer = SprintSerializer(data=request.data)
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+                return Response(
+                    {
+                        "message": "Sprint Added Successfully",
+                        "data": serializer.data
+                    }, status=status.HTTP_201_CREATED
+                )
+            return Response(
+                {
+                    "message": "Only admin or superuser can add a sprint"
+
+                }, status=status.HTTP_401_UNAUTHORIZED
+            )
+        except InsertionError as e:
+            return Response(
+                {
+                    "message": "Error Occurred",
+                    "error": str(e)
+                }, status=status.HTTP_400_BAD_REQUEST
+            )
+        except Exception as e:
+            return Response(
+                {"message": "Error Occurred",
+                 "error": str(e)
+                 }, status=status.HTTP_500_INTERNAL_SERVER_ERROR
+            )
+
+    @verify_token
+    def put(self, request, id):
+        """
+        For Updating the Sprint Cycle
+        :param request:
+        :return:
+        """
+        try:
+            user = User.objects.get(id=request.data.get("user_id"))
+            if user.is_superuser:
+                sprint = Sprint.objects.get(id=id)
```

Comment:
if sprint with given id doesn't exist, this code will raise an exception which will be caught at the bottom and give 500 internal server error. Instead, we should show an appropriate error message like 404 sprint not found.

Reply:
Added the code for checking sprint exist or not.